### PR TITLE
Bluetooth: controller: Refactor adv random delay resolution fix

### PR DESCRIPTION
**Pull request by cvinayak**
Refactored the fix in the commit 685da0235446 ("Bluetooth:
controller: Fix advertising random delay resolution calc")
to apply modulo in tick units before adding a tick.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by cvinayak** [comment]
@carlescufi lets wait for @mike-scott approval.

**Review by carlescufi — APPROVED** [review]

**Comment by mike-scott** [comment]
Apologies for my time-zone.  Looking at this now.

**Comment by mike-scott** [comment]
@cvinayak I'm curious what the expected range is supposed to be?  Using this patch 328 ticks is the maximum result (+1).

Under the older code, I think we were getting higher values.

**Comment by mike-scott** [comment]
Changes look good to me, btw.

**Review by mike-scott — APPROVED** [review]

**Pull request merged**
